### PR TITLE
Zosiu's layout for the keyboardio atreus

### PR DESCRIPTION
**Pull request by zosiu**
## Description

Add my keyboardio atreus keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c [hunk 1]
```diff
@@ -0,0 +1,265 @@
+#include QMK_KEYBOARD_H
+
+#define USB_MAX_POWER_CONSUMPTION 100
+#define PREVENT_STUCK_MODIFIERS
+
```
Comment (zvecr) [inline keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c:5]: These wont do anything here, and would need to go in a `config.h`
```suggestion
```

File: keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+};
+
+void eeconfig_init_user(void) { set_unicode_input_mode(UC_LNX); }
+
+void persistant_default_layer_set(uint16_t default_layer) {
+    eeconfig_update_default_layer(default_layer);
+    default_layer_set(default_layer);
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt) {
```
Comment (zvecr) [inline keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c:139]: `action_get_macro` has been deprecated, and needs to be migrated to <https://docs.qmk.fm/#/feature_macros?id=the-new-way-send_string-amp-process_record_user>

**Review by zvecr — CHANGES_REQUESTED** [review]
You should also add a licence header to any `*.c` or `*.h` files

File: keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c (hunk 2 above)
Comment (zosiu) [reply keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c:143]: I migrated one part of it. The activation/deactivation of the oneshot layer however wouldn't work in `process_record_user` no matter how I tried, so I left that as is for now.

**Review by noroadsleft — APPROVED** [review]
Slight `__attribute__ ((weak))` :heavy_check_mark: as I don't use One-Shots, but looks good otherwise.

File: keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+  ),
+
+  [_MOD_S] = LAYOUT( \
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    XXXXXXX, _______, XXXXXXX, KC_SI,   XXXXXXX,                   XXXXXXX, KC_SI,   XXXXXXX, XXXXXXX, XXXXXXX,
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
+  ),
+
+  [_MOD_Z] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c:272]:
```suggestion
  [_WM] = LAYOUT(
    KC_Q,    KC_D,    KC_R,    KC_W,    KC_B,                      KC_J,    KC_F,    KC_U_WD, KC_P,    KC_SCLN,
    KC_A_WD, KC_S_WD, KC_H,    KC_T,    KC_G,                      KC_Y,    KC_N_WD, KC_E_WD, KC_O_WD, KC_I_WD,
    KC_Z_WD, KC_X,    KC_M,    KC_C_WD, KC_V,    XXXXXXX, XXXXXXX, KC_K,    KC_L_WD, DK_ACT,  KC_DOT,  KC_SLSH,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_SHTB, KC_CTES, KC_ENOP, KC_BSCM, KC_SPFN, KC_SHTB, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_DK] = LAYOUT(
    KC_PERC, KC_AMPR, KC_QUES, KC_PLUS, KC_AT,                     KC_DLR,  KC_UNDS, KC_LBRC, KC_RBRC, KC_EXLM,
    KC_HASH, KC_LPRN, KC_EQL,  KC_0,    KC_LCBR,                   KC_RCBR, KC_1,    KC_ASTR, KC_RPRN, KC_MINS,
    KC_6,    KC_7,    KC_8,    KC_9,    KC_PIPE, XXXXXXX, XXXXXXX, KC_BSLS, KC_2,    DK_REL,  KC_4,    KC_5,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_SHTB, KC_CTES, KC_ENOP, KC_BSCM, KC_COMS, KC_SHTB, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_FUN] = LAYOUT(
    XXXXXXX, KC_DEL,  KC_C_LF, KC_C_RT, KC_INS,                    KC_MUTE, KC_VOLD, KC_VOLU, KC_MPLY, KC_PSCR,
    KC_GRV,  KC_END,  KC_PGUP, KC_PGDN, KC_HOME,                   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_QUOT,
    XXXXXXX, KC_CSTB, KC_C_TB, KC_COPY, KC_PSTE, XXXXXXX, XXXXXXX, XXXXXXX, KC_BRID, KC_BRIU, XXXXXXX, KC_CIRC,
    UC_M_OS, UC_M_LN, UC_M_WC, KC_SHTB, XXXXXXX, KC_ENOP, KC_BSPC, _______, KC_SHTB, XXXXXXX, RESET,   CG_TOGG
  ),

  [_MOD_A] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    _______, XXXXXXX, XXXXXXX, KC_AA,   XXXXXXX,                   XXXXXXX, KC_AA,   XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_AU,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_AU,   XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_I] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_II  , XXXXXXX,                   XXXXXXX, KC_II,   XXXXXXX, XXXXXXX, _______,
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_U] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, KC_UEE,  XXXXXXX,                   XXXXXXX, XXXXXXX, _______, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_UU,   XXXXXXX,                   XXXXXXX, KC_UU,   XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_UE,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_UE,   KC_UEE,  XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_E] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_EE,   XXXXXXX,                   XXXXXXX, KC_EE,   _______, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_EU,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_EU,   XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_O] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, KC_OEE,  XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_OO,   XXXXXXX,                   XXXXXXX, KC_OO,   XXXXXXX, _______, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_OE,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_OE,   KC_OEE,  XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_C] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, KC_CI,   XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_L] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_EL,   XXXXXXX,                   XXXXXXX, KC_EL,   XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, _______, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_N] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_NI,   XXXXXXX,                   XXXXXXX, _______, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_S] = LAYOUT(
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, _______, XXXXXXX, KC_SI,   XXXXXXX,                   XXXXXXX, KC_SI,   XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
    XXXXXXX, XXXXXXX, XXXXXXX, KC_LSFT, XXXXXXX, KC_ENOP, XXXXXXX, XXXXXXX, KC_RSFT, XXXXXXX, XXXXXXX, XXXXXXX
  ),

  [_MOD_Z] = LAYOUT(
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keyboardio/atreus/keymaps/zosiu/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Zosiu's atreus layout
```
[the remark this answers is not in the document]
Comment (zosiu) [reply keyboards/keyboardio/atreus/keymaps/zosiu/README.md]: done!

**Review by noroadsleft — APPROVED** [review]

File: keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c (hunk 2 above)
Comment (drashna) [reply keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c:143]: Try using `post_process_record_user` and see if that works better. 

If that works, then I think I know what the issue is, then. :(

File: keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c (hunk 3 above)
Comment (fauxpark) [reply keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c:272]: This is still outstanding.

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

**Review by fauxpark — CHANGES_REQUESTED** [review]
The `action_get_macro` stuff still needs to be removed/migrated.

**Comment by zosiu** [comment]
> The `action_get_macro` stuff still needs to be removed/migrated.

I'm closing this then because I was unable to get rid of `action_get_macro` while keeping things the same as before.

**Pull request closed without merging**